plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveWriteUtils.java

```diff
@@ -453,7 +453,9 @@ public static Location getTableDefaultLocation(Database database, HdfsContext co
             }
         }
 
-        return Location.of(location).appendPath(escapeTableName(tableName));
+        // Note: this results in a "normalized location", e.g. not containing double slashes.
+        // TODO (https://github.com/trinodb/trino/issues/17803): We need to use normalized location until all relevant Hive connector components are migrated off Hadoop Path.
+        return Location.of(databasePath.toString()).appendPath(escapeTableName(tableName));
```

Comment:
```
```
trino> create schema hive.doubleslashes with (location='s3://trino-test/double_slash//location');
CREATE SCHEMA
trino> show create schema hive.doubleslashes;
                         Create Schema                          
----------------------------------------------------------------
 CREATE SCHEMA hive.doubleslashes                               
 WITH (                                                         
    location = 's3://trino-test/double_slash//location' 
 )                                                              
(1 row)

trino> create table hive.doubleslashes.test1 (c integer);
CREATE TABLE
trino> insert into hive.doubleslashes.test1 values 1,2,3;
INSERT: 3 rows
sele
Query 20230620_060740_00002_8c4bz, FINISHED, 1 node
http://localhost:8080/ui/query.html?20230620_060740_00002_8c4bz
Splits: 14 total, 14 done (100.00%)
CPU Time: 0.3s total,     0 rows/s,     0B/s, 28% active
Per Node: 0.1 parallelism,     0 rows/s,     0B/s
Parallelism: 0.1
Peak Memory: 38.2KB
3.11 [0 rows, 0B] [0 rows/s, 0B/s]

trino> select "$path"  from hive.doubleslashes.test1;
                                                        $path                                                         
----------------------------------------------------------------------------------------------------------------------
 s3://trino-test/double_slash/location/test1/20230620_060740_00002_8c4bz_e91a2ac5-4bc1-40ba-ab3f-2145eaa6bf97 
 s3://trino-test/double_slash/location/test1/20230620_060740_00002_8c4bz_e91a2ac5-4bc1-40ba-ab3f-2145eaa6bf97 
 s3://trino-test/double_slash/location/test1/20230620_060740_00002_8c4bz_e91a2ac5-4bc1-40ba-ab3f-2145eaa6bf97 
(3 rows)
```

Note that this strategy may eventually cause issues with _managed_ tables which get created into schemas having a double-slashed location.
```

Reply:
> schema location: `s3://trino-test/double_slash//location`
> table location: `s3://trino-test/double_slash/location/test1`

well, that's the whole point of this PR

thanks for calling this out!